File: Prime/Recursion/Basic/fibonaci.js

// 0 1 1 2 3 5

// recursion way
function fibonaci(n){

  if(n < 2){
    return n;
  }

  const result = fibonaci(n-1) + fibonaci(n-2);
  return result;
  
}
// console.log(fibonaci(5));

export default fibonaci;


// itrative way
function fibonaciWithItration(n){
  let prev = 0;
  let curr = 1;
  n -= 2;
  while(n != 0){
    next = prev + curr;
    prev = curr;
    curr = next;
    n--;
  }

  return next;
}

// console.log(fibonaciWithItration(50));

File: Recursion/Basic/Fibonaci.java
class Fibonaci {

  public static int recursive(int n) {
    if (n == 0 || n == 1) {
      return n;
    }

    return recursive(n - 1) + recursive(n - 2);
  }

  public static void main(String[] args) {
    System.out.println(recursive(6));
  }
}
